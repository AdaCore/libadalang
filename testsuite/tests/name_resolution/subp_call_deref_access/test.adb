procedure Test is
   type Access_Proc is access procedure (Flag : Boolean);

   function Func return Access_Proc is (null);

   X : Access_Proc;
begin
   X := Func.all'Access;
   pragma Test_Statement;
end Test;
