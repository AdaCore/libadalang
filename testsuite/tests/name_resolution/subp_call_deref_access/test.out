Analyzing test.adb
##################

Resolving xrefs for node <AssignStmt test.adb:8:4-8:25>
*******************************************************

Expr: <Id "X" test.adb:8:4-8:5>
  references:    <DefiningName "X" test.adb:6:4-6:5>
  type:          <ConcreteTypeDecl ["Access_Proc"] test.adb:2:4-2:58>
  expected type: None
Expr: <AttributeRef test.adb:8:9-8:24>
  references:    None
  type:          <ConcreteTypeDecl ["Access_Proc"] test.adb:2:4-2:58>
  expected type: <ConcreteTypeDecl ["Access_Proc"] test.adb:2:4-2:58>
Expr: <ExplicitDeref test.adb:8:9-8:17>
  references:    <DefiningName "Func" test.adb:4:13-4:17>
  type:          None
  expected type: None
Expr: <Id "Func" test.adb:8:9-8:13>
  references:    <DefiningName "Func" test.adb:4:13-4:17>
  type:          <ConcreteTypeDecl ["Access_Proc"] test.adb:2:4-2:58>
  expected type: <ConcreteTypeDecl ["Access_Proc"] test.adb:2:4-2:58>
Expr: <Id "Access" test.adb:8:18-8:24>
  references:    None
  type:          None
  expected type: None


Done.
